diff --git a/Makefile.am b/Makefile.am
index 14dc830..a8dde9d 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -59,20 +59,11 @@ jack_mix_box_LDADD = $(JACKMIXER_LIBS) -lm
 test: _jack_mixer_c.so
 	@./test.py
 
-schemadir = @GCONF_SCHEMA_FILE_DIR@
-schema_DATA = jack_mixer.schemas
-
 install-exec-hook:
 	ln -vfs jack_mixer.py $(DESTDIR)$(bindir)/jack_mixer
 
-if GCONF_SCHEMAS_INSTALL
-install-data-hook:
-	GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(srcdir)/$(schema_DATA)
-
 uninstall-hook:
-	GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-uninstall-rule $(srcdir)/$(schema_DATA)
 	rm -vf $(DESTDIR)$(bindir)/jack_mixer
-endif
 
 pacoinstall:
 	test `whoami` == root # test for root
